centrifuge-app/src/components/Report/CashflowStatement.tsx

```diff
@@ -0,0 +1,287 @@
+      return [from, to]
+    } else if (groupBy === 'daily') {
+      const from = new Date(startDate ?? today)
+      from.setHours(0, 0, 0, 0)
+      const to = new Date(endDate ?? today)
+      to.setDate(to.getDate() + 1)
+      to.setHours(0, 0, 0, 0)
+      return [from, to]
+    } else if (groupBy === 'quarter' || groupBy === 'year') {
+      const from = pool.createdAt ? new Date(pool.createdAt) : today
+      return [from, today]
+    } else {
+      const to = new Date(endDate ?? today)
+      to.setDate(to.getDate() + 1)
+      to.setHours(0, 0, 0, 0)
+      return [new Date(startDate), to]
+    }
+  }, [groupBy, startDate, endDate])
+
+  const poolStates = useAggregatedPoolStatesByGroup(
+    pool.id,
+    adjustedStartDate,
+    adjustedEndDate,
+    groupBy === 'daily' ? 'day' : groupBy
+  )
+
+  const poolStatesNotAggregated = usePoolStatesByGroup(
+    pool.id,
+    adjustedStartDate,
+    adjustedEndDate,
+    groupBy === 'daily' ? 'day' : groupBy
+  )
+
+  const columns = React.useMemo(() => {
+    if (!poolStates || !poolStates.length) {
+      return []
+    }
+
+    const getColumnHeader = (timestamp: string) => {
+      if (groupBy === 'day' || groupBy === 'daily') {
+        return new Date(timestamp).toLocaleDateString('en-US', {
+          day: 'numeric',
+          month: 'short',
+          year: 'numeric',
+        })
+      } else if (groupBy === 'month') {
+        return new Date(timestamp).toLocaleDateString('en-US', {
+          month: 'long',
+          year: 'numeric',
+        })
+      } else if (groupBy === 'quarter') {
+        const date = new Date(timestamp)
+        return `Q${Math.floor(date.getMonth() / 3) + 1} ${date.getFullYear()}`
+      } else if (groupBy === 'year') {
+        return new Date(timestamp).toLocaleDateString('en-US', {
+          year: 'numeric',
+        })
+      }
+      return ''
+    }
+
+    return [
+      {
+        align: 'left',
+        header: '',
+        cell: (row: Row) => (
+          <Text variant={row.heading ? 'heading4' : row.bold ? 'interactive2' : 'body3'}>{row.name}</Text>
+        ),
+        width: '200px',
+      },
+    ]
+      .concat(
+        poolStates.map((state, index) => ({
+          align: 'right',
+          timestamp: state.timestamp,
+          header: getColumnHeader(state.timestamp),
+          cell: (row: Row) => (
+            <Text variant={row.heading ? 'heading4' : row.bold ? 'interactive2' : 'body3'}>
+              {row.formatter ? row.formatter((row.value as any)[index]) : (row.value as any)[index]}
+            </Text>
+          ),
+          width: '170px',
+        }))
+      )
+      .concat({
+        align: 'left',
+        header: '',
+        cell: () => <span />,
+        width: '1fr',
+      })
+  }, [poolStates, groupBy, pool])
+
+  const grossCashflowRecords: Row[] = React.useMemo(() => {
+    return [
+      {
+        name: poolMetadata?.pool?.asset.class === 'privateCredit' ? 'Repayments' : 'Sales',
```

Comment:
This doesn't seem to work, I think it needs to be
```suggestion
        name: poolMetadata?.pool?.asset.class === 'Private credit' ? 'Repayments' : 'Sales',
```

Reply:
You can test this on the NS3 pool, see https://pr2181-app-ff-production.k-f.dev/pools/1615768079/reporting/cashflow-statement